url7f43a4752f_shalynnho_pokemonfactory_tgz-pJ8-Networking_ServerJ8/src/agent/CameraAgent.java

```diff
@@ -0,0 +1,554 @@
+
+	private final String name;
+
+	public List<MyNest> nests = Collections.synchronizedList(new ArrayList<MyNest>());
+	public MyKit mk;
+
+	public CameraGraphics guiCamera;
+	private Timer timer;
+
+	public KitRobotAgent kitRobot;
+	public PartsRobotAgent partRobot;
+
+	Semaphore animation = new Semaphore(0, true);
+
+	public enum NestStatus {
+		NOT_READY, STARTED_TIMER, READY, PHOTOGRAPHING, PHOTOGRAPHED, NEEDS_TO_PURGE, WAITING_TO_RE_PHOTOGRAPH, READY_TO_RE_PHOTOGRAPH, RE_PHOTOGRAPHING_ONCE, RE_PHOTOGRAPHED_ONCE, NEED_TO_RE_PHOTOGRAPH_AGAIN, WAITING_TO_RE_PHOTOGRAPH_AGAIN, READY_TO_RE_PHOTOGRAPH_AGAIN, RE_PHOTOGRAPHING_TWICE, RE_PHOTOGRAPHED_TWICE
+	};
+
+	public enum KitStatus {
+		NOT_READY, DONE, PICTURE_BEING_TAKEN, FAILED
+	};
+
+	public class MyKit {
+		public Kit kit;
+		public KitStatus ks;
+		public boolean kitDone;
+
+		public MyKit(Kit k) {
+			kit = k;
+			ks = KitStatus.NOT_READY;
+			kitDone = false;
+		}
+	}
+
+	public class MyNest {
+		public NestAgent nest;
+		public NestStatus state;
+		public int numFilledSnapshot = 0;
+
+		public MyNest(NestAgent nest) {
+			this.nest = nest;
+			this.state = NestStatus.NOT_READY;
+		}
+	}
+
+	public CameraAgent(String name) {
+		super();
+		this.name = name;
+
+		timer = new Timer();
+		timer.scheduleAtFixedRate(new TimerTask() {
+			// Fires every 3.001 seconds.
+			@Override
+			public void run() {
+				print("Waking up");
+				stateChanged();
+			}
+		}, System.currentTimeMillis(), 3000);
+	}
+
+	/*
+	 * Messages
+	 */
+	@Override
+	public void msgInspectKit(Kit kit) {
+		print("Received msgInspectKit");
+		mk = new MyKit(kit);
+		stateChanged();
+	}
+
+	// Should no longer be called in v2. Change to timer.
+	@Override
+	public void msgIAmFull(Nest nest) {
+		// synchronized (nests) {
+
+		// print("Received msgIAmFull from nest with type " + ((NestAgent)
+		// nest).currentPartType);
+		for (MyNest n : nests) {
+			if (n.nest == nest) {
+				if (n.state == NestStatus.NOT_READY || n.state == NestStatus.STARTED_TIMER) {
+					n.state = NestStatus.READY;
+				} else if (n.state == NestStatus.WAITING_TO_RE_PHOTOGRAPH) {
+					n.state = NestStatus.READY_TO_RE_PHOTOGRAPH;
+				} else if (n.state == NestStatus.WAITING_TO_RE_PHOTOGRAPH_AGAIN) {
+					n.state = NestStatus.READY_TO_RE_PHOTOGRAPH_AGAIN;
+				}
+				break;
+			}
+		}
+		// }
+		stateChanged();
+	}
+
+	@Override
+	public void msgResetSelf() {
+		print("Reseting Self");
+		// synchronized (nests) {
+		for (MyNest n : nests) {
+			n.state = NestStatus.NOT_READY;
+		}
+		// }
+		mk = null;
+		timer.cancel();
+		timer = new Timer();
+		timer.scheduleAtFixedRate(new TimerTask() {
+			// Fires every 3.001 seconds.
+			@Override
+			public void run() {
+				print("Waking up");
+				stateChanged();
+			}
+		}, System.currentTimeMillis(), 3000);
+		// stateChanged();
+	}
+
+	@Override
+	public void msgTakePictureNestDone(NestGraphics nest, boolean done, NestGraphics nest2, boolean done2) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.